packages/runtime/src/helpers/functions.ts

```diff
@@ -364,58 +369,154 @@ const changeExtension = (file: string, extension: string) => {
+      const pageData =
+        routerTypeSubPath === 'app'
+          ? await readFile(dataFilePath, 'utf8')
+          : JSON.parse(await readFile(join(publish, 'server', routerTypeSubPath, `${route}.json`), 'utf8'))
 
-      if (ssgRoute.dataRoute.endsWith('.rsc')) {
-        return [
-          join(publish, 'server', 'app', ssgRoute.dataRoute),
-          join(publish, 'server', 'app', changeExtension(ssgRoute.dataRoute, '.html')),
-        ]
+      const htmlFilePath = join(publish, 'server', routerTypeSubPath, `${route}.html`)
+      const html = await readFile(htmlFilePath, 'utf8')
+
+      return {
+        key: route,
+        data: {
+          value: {
+            kind: 'PAGE' as const,
+            html,
+            pageData,
+          },
+          lastModified: Date.now(),
+        },
       }
+    }),
+  )
+
+  const prerenderedContent = await Promise.all(readFilePromises)
+  return prerenderedContent
+}
+
+const getPrerenderedContent = (prerenderManifest: PrerenderManifest, publish: string): string[] => [
+  ...Object.entries(prerenderManifest.routes).flatMap(([route, ssgRoute]) => {
+    if (ssgRoute.initialRevalidateSeconds === false) {
+      return []
+    }
 
-      const trimmedPath = route === '/' ? 'index' : route.slice(1)
+    if (ssgRoute.dataRoute.endsWith('.rsc')) {
       return [
-        join(publish, 'server', 'pages', `${trimmedPath}.html`),
-        join(publish, 'server', 'pages', `${trimmedPath}.json`),
+        join(publish, 'server', 'app', ssgRoute.dataRoute),
+        join(publish, 'server', 'app', changeExtension(ssgRoute.dataRoute, '.html')),
       ]
-    }),
-    join(publish, '**', '*.html'),
-    join(publish, 'static-manifest.json'),
-  ]
+    }
+
+    const trimmedPath = route === '/' ? 'index' : route.slice(1)
+    return [
+      join(publish, 'server', 'pages', `${trimmedPath}.html`),
+      join(publish, 'server', 'pages', `${trimmedPath}.json`),
+    ]
+  }),
+  join(publish, '**', '*.html'),
+  join(publish, 'static-manifest.json'),
+]
+
+type EnhancedNetlifyPluginConstants = NetlifyPluginConstants & {
+  NETLIFY_API_HOST?: string
+  NETLIFY_API_TOKEN?: string
 }
 
-export const getSSRLambdas = async (publish: string): Promise<SSRLambda[]> => {
+// TODO: get a build feature flag set up for blob storage
+export const getSSRLambdas = async ({
+  publish,
+  constants,
+  featureFlags,
+}: {
+  publish: string
+  constants: EnhancedNetlifyPluginConstants
+  featureFlags: Record<string, unknown>
+}): Promise<SSRLambda[]> => {
+  console.dir(featureFlags)
   const commonDependencies = await getCommonDependencies(publish)
   const ssrRoutes = await getSSRRoutes(publish)
 
   // TODO: for now, they're the same - but we should separate them
   const nonOdbRoutes = ssrRoutes
   const odbRoutes = ssrRoutes
+  const { NETLIFY_API_HOST, NETLIFY_API_TOKEN, SITE_ID } = constants
+
+  // This check could be improved via featureFlags exposed in the build for blob storage
+  const isUsingBlobStorage =
```

Comment:
I added checks for the deploy ID because blob storage will not be available if you run only `ntl build`, but it will be available once @eduardoboucas makes some changes for blob storage to be supported via `ntl deploy`.